receiver/apachesparkreceiver/metadata.yaml

```diff
@@ -4,4 +4,783 @@ status:
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: bytes
+    attributes:
+      [application_id, application_name, stage_id, attempt_id, stage_status]
+  spark.stage.jvm_gc_time:
+    description: The amount of time the JVM spent on garbage collection in this stage.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: bytes
+    attributes:
+      [application_id, application_name, stage_id, attempt_id, stage_status]
+  spark.stage.memory_spilled:
+    description: The amount of memory moved to disk due to size constraints (spilled) in this stage.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: bytes
+    attributes:
+      [application_id, application_name, stage_id, attempt_id, stage_status]
+  spark.stage.disk_space_spilled:
+    description: The amount of disk space used for storing portions of overly large data chunks that couldn’t fit in memory in this stage.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: bytes
+    attributes:
+      [application_id, application_name, stage_id, attempt_id, stage_status]
+  spark.stage.peak_execution_memory:
+    description: Peak memory used by internal data structures created during shuffles, aggregations and joins in this stage.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: bytes
+    attributes:
+      [application_id, application_name, stage_id, attempt_id, stage_status]
+  spark.stage.input_bytes:
+    description: Number of bytes read in this stage.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: bytes
+    attributes:
+      [application_id, application_name, stage_id, attempt_id, stage_status]
+  spark.stage.input_records:
+    description: Number of records read in this stage.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: "{ records }"
+    attributes:
+      [application_id, application_name, stage_id, attempt_id, stage_status]
+  spark.stage.output_bytes:
+    description: Number of bytes written in this stage.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: bytes
+    attributes:
+      [application_id, application_name, stage_id, attempt_id, stage_status]
+  spark.stage.output_records:
+    description: Number of records written in this stage.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: "{ records }"
+    attributes:
+      [application_id, application_name, stage_id, attempt_id, stage_status]
```

Comment:
I suggest grouping these into two metrics that share a namespace: `spark.stage.io.records` and `spark.stage.io.size`

Both can use the same attribute, e.g. `direction = ( in | out )`

Reply:
I brought this up with Sam during the spec writing, and he said it was a divisive topic in the OTel community. I'll do the refactor though since it cuts down on metric bloat.